### PR TITLE
Update URL creation to be more robust.

### PR DESCRIPTION
**Pull request by KevinJoiner**

### CODE REVIEW

File: internal/repositories/aftermarket/aftermarket_devices.go [hunk 1]
```diff
@@ -209,6 +219,11 @@ func ToAPI(d *models.AftermarketDevice, imageUrl string) (*gmodel.AftermarketDev
 	}, nil
 }
 
+func GetAftermarketDeviceImageUrl(baseURL string, tokenID int) (string, error) {
```
Comment (0xdev22) [inline internal/repositories/aftermarket/aftermarket_devices.go:222]: Can you add a test for this too the way we have a test for `GetVehicleImageUrl`

**Review by 0xdev22 — CHANGES_REQUESTED** [review]

**Review by Allyson-English — APPROVED** [review]

File: internal/loader/aftermarket_device_loader.go [hunk 2]
```diff
@@ -56,11 +57,18 @@ func (ad *AftermarketDeviceLoader) BatchGetLinkedAftermarketDeviceByVehicleID(ct
 
 	for i, vID := range vehicleIDs {
 		if am, ok := amByVehicleID[vID]; ok {
-			imageUrl := helpers.GetAftermarketDeviceImageUrl(ad.settings.BaseImageURL, am.ID)
+			var retErr error
+			imageUrl, err := aftermarket.GetAftermarketDeviceImageUrl(ad.settings.BaseImageURL, am.ID)
```
Comment (elffjs) [inline internal/loader/aftermarket_device_loader.go:61]:
```suggestion
			imageURL, err := aftermarket.GetAftermarketDeviceImageUrl(ad.settings.BaseImageURL, am.ID)
```

Similarly in other places.

It's weird that this call can fail. I guess the culprit is [`url.Parse`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.2:src/net/url/url.go;l=1266).

File: internal/repositories/aftermarket/aftermarket_devices.go [hunk 3]
```diff
@@ -173,7 +179,11 @@ func (r *Repository) GetAftermarketDevice(ctx context.Context, by gmodel.Afterma
 		return nil, err
 	}
 
-	imageUrl := helpers.GetAftermarketDeviceImageUrl(r.Settings.BaseImageURL, ad.ID)
+	imageUrl, err := GetAftermarketDeviceImageUrl(r.Settings.BaseImageURL, ad.ID)
+	if err != nil {
+		return nil, fmt.Errorf("faile to get image url: %w", err)
```
Comment (elffjs) [inline internal/repositories/aftermarket/aftermarket_devices.go:184]:
```suggestion
		return nil, fmt.Errorf("failed to get image url: %w", err)
```

**Review by elffjs — CHANGES_REQUESTED** [review]
Minor things.

File: internal/loader/aftermarket_device_loader.go (hunk 2 above)
Comment (elffjs) [reply internal/loader/aftermarket_device_loader.go:61]: Is it resolved? I think Go casing is `imageURL`.

Comment (elffjs) [reply internal/loader/aftermarket_device_loader.go:61]: I made a PR into your PR #94

**Review by elffjs — APPROVED** [review]

**Comment by elffjs** [comment]
Good to merge, I think?

**Pull request merged**